Documentation/ApiOverview/DependencyInjection/Index.rst

```diff
@@ -177,39 +177,48 @@ Public
 
 :yaml:`public: false` is a performance optimization and is therefore suggested to be
 set in extensions. This settings controls which services are available
-through :php:`\Psr\Container\ContainerInterface->get()`. However some classes, that need to
-be public, will be marked public automatically due to :yaml:`autoconfigure: true`.
-These classes include Singletons, because they need to be shared with code that uses
-:php:`\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance()` and Extbase controllers.
+through the dependency injection container which is used internally by
+:php:`GeneralUtility::makeInstance()`.
+
+However some classes that need to be public will be marked public automatically
+due to :yaml:`autoconfigure: true`.
+
+These classes include singletons, because they need to be shared with code that uses
+:php:`GeneralUtility::makeInstance()` and Extbase controllers.
 
 .. index:: Dependency injection; Public
 .. _knowing-what-to-make-public:
 
 Knowing what to make public
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Instances of :php:`\TYPO3\CMS\Core\SingletonInterface` and Extbase controllers are
-marked public by default. Additionally some classes cannot be private as well.
-As the Symfony documentation puts it:
 
 .. this indent is intentional to create a blockquote!
 
     "Simply said: A service can be marked as private if you do not want to access it directly from your code."
 
     -- `Official documentation <https://symfony.com/doc/current/service_container/alias_private.html>`_ for public and private services.
 
-Direct access includes instantiation via :php:`\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance()`.
-This means every class that should make use of dependency injection and is not instantiated via injection
-itself but by e.g. :php:`\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance()` have to be marked
-as public. Some examples for this are:
+Direct access includes instantiation via :php:`GeneralUtility::makeInstance()` with constructor arguments.
+
+This means every class that needs dependency injection and is retrieved directly, e.g.
+using :php:`GeneralUtility::makeInstance()` must be marked as public.
+
+Any other class which needs dependency injection and is retrieved by dependency injection
+itself can be private.
+
+Instances of :php:`\TYPO3\CMS\Core\SingletonInterface` and Extbase controllers are
+automatically marked as public since they are retrieved using :php:`GeneralUtility::makeInstance()`.
```

Comment:
> since they are retrieved using :php:`GeneralUtility::makeInstance()`
I think  **because** would be better here

Reply:
Indeed, _since_ can be understood as time reference, too.

Reply:
Or rather even
"because they shall be retrievable using :php:GeneralUtility::makeInstance()"

I just wanted to make clear that the system does not make them public just by being used like that. The system has been designed to automatically make them public for convenience because that is a very common use case.